src/helium_follower_service.erl

```diff
@@ -41,21 +44,97 @@ txn_stream(_Msg, StreamState) ->
+    case Chain of
+        undefined ->
+            lager:debug("chain not ready, returning error response for msg ~p", [_Msg]),
+            {grpc_error, {grpcbox_stream:code_to_status(14), <<"temporarily unavailable">>}};
+        _ ->
+            Ledger = blockchain:ledger(Chain),
+            {ok, Height} = blockchain_ledger_v1:current_height(Ledger),
+
+            InteractiveBlock = case ?get_var(?hip17_interactivity_blocks, Ledger) of
+                                   {ok, V} -> V;
+                                   {error, not_found} -> 0
+                               end,
+            {NumGateways, FinalGws, StreamState1} =
+                blockchain_ledger_v1:cf_fold(
+                    active_gateways,
+                    fun({Addr, BinGw}, {CountAcc, GwAcc, StreamAcc}) ->
+                        Gw = blockchain_ledger_gateway_v2:deserialize(BinGw),
+                        Loc = blockchain_ledger_gateway_v2:location(Gw),
+                        LastBeacon = case blockchain_ledger_v1:find_gateway_last_beacon(Addr, Ledger) of
+                                         {ok, LB} -> LB;
+                                         {error, _} -> undefined
+                                     end,
+                        case Loc /= undefined andalso (LastBeacon /= undefined andalso (Height - LastBeacon) =< InteractiveBlock) of
+                            true ->
+                                Region = case blockchain_region_v1:h3_to_region(Loc, Ledger) of
+                                             {ok, R} -> normalize_region(R);
+                                             _ -> undefined
+                                         end,
+                                GwResp = #follower_gateway_resp_v1_pb{
+                                          height = Height,
+                                          result = {info, #gateway_info_pb{
+                                              address = Addr,
+                                              location = h3:to_string(Loc),
+                                              owner = blockchain_ledger_gateway_v2:owner_address(Gw),
+                                              staking_mode = blockchain_ledger_gateway_v2:mode(Gw),
+                                              gain = blockchain_ledger_gateway_v2:gain(Gw),
+                                              region = Region
+                                      }}},
+                                GwAcc1 = [GwResp | GwAcc],
+                                RespLen = length(GwAcc1),
+                                case RespLen >= ?GW_STREAM_BATCH_SIZE of
+                                    true ->
+                                        Resp = #follower_gateway_stream_resp_v1_pb{ gateways = GwAcc1 },
+                                        StreamAcc1 = grpcbox_stream:send(false, Resp, StreamAcc),
+                                        timer:sleep(100),
```

Comment:
why is there a sleep there?

Reply:
crap, debugging step i missed removing; fixed [9a49c5a](https://github.com/helium/blockchain-node/pull/178/commits/9a49c5a116a3502ffcf3301e2c01b1de9d1e884e)